src/lsstseries/ensemble.py

```diff
@@ -391,6 +408,69 @@ def batch(self, func, *args, meta=None, use_map=True, compute=True, on=None, **k
         else:
             return batch
 
+    def from_hipscat(
+        self,
+        dir,
+        source_subdir="source",
+        object_subdir="object",
+        id_col=None,
+        time_col=None,
+        flux_col=None,
+        err_col=None,
+        band_col=None,
+        additional_cols=True,
+    ):
+        """Read in parquet files from a hipscat-formatted directory structure
+        Parameters
+        ----------
+        dir: 'str'
+            Path to the directory structure
+        source_subdir: 'str'
+            Path to the subdirectory which contains source files
+        object_subdir: 'str'
+            Path to the subdirectory which contains object files, if None then
+            files will only be read from the source_subdir
+        id_col: 'str', optional
+            Identifies which column contains the Object IDs
+        time_col: 'str', optional
+            Identifies which column contains the time information
+        flux_col: 'str', optional
+            Identifies which column contains the flux/magnitude information
+        err_col: 'str', optional
+            Identifies which column contains the flux/mag error information
+        band_col: 'str', optional
+            Identifies which column contains the band information
+        additional_cols: 'bool', optional
+            Boolean to indicate whether to carry in columns beyond the
+            critical columns, true will, while false will only load the columns
+            containing the critical quantities (id,time,flux,err,band)
+
+        Returns
+        ----------
+        ensemble: `lsstseries.ensemble.Ensemble`
+            The ensemble object with parquet data loaded
+        """
+
+        source_path = os.path.join(dir, source_subdir)
+        source_files = glob.glob(os.path.join(source_path, "**", "*.parquet"), recursive=True)
+
+        if object_subdir is not None:
+            object_path = os.path.join(dir, object_subdir)
+            object_files = glob.glob(os.path.join(object_path, "**", "*.parquet"), recursive=True)
+        else:
+            object_files = None
+
+        return self.from_parquet(
+            source_files,
```

Comment:
Should you pass in provenance information here? A natural option would be something like `source_subdir` or adding a provenance parameter to the `from_hipscat` call.

Reply:
Ah, yes good catch. But we may want to point to provenance from hipscat. Just looking through their provenance.json files, they seem to indicate properties of the data generation, rather than a single label to give the whole set. We could even by default just use the top level dir as a provenance label (would be "ztf_mar16" in this case)

Reply:
Makes sense.